pkg/storage/tsdb/bucketindex/updater.go

```diff
@@ -123,10 +132,13 @@ func (w *Updater) updateBlockIndexEntry(ctx context.Context, id ulid.ULID) (*Blo
 	metaFile := path.Join(id.String(), block.MetaFilename)
 
 	// Get the block's meta.json file.
-	r, err := w.bkt.ReaderWithExpectedErrs(w.bkt.IsObjNotFoundErr).Get(ctx, metaFile)
+	r, err := w.bkt.ReaderWithExpectedErrs(tsdb.IsObjNotFoundOrCustomerManagedKeyErr(w.bkt)).Get(ctx, metaFile)
```

Comment:
nit: I wonder if we can generalize `tsdb.IsObjNotFoundOrCustomerManagedKeyErr` a bit by accepting list errors as argument. like 

`tsdb.IsExpectedErrors(w.bkt, bkt.IsObjNotFoundErr, bkt.IsCustomerManagedKeyErr)`